pypeit/spectrographs/keck_lris.py

```diff
@@ -168,7 +178,14 @@ def compound_meta(self, headarr, meta_key):
             return binning
         elif 'lampstat' in meta_key:
             idx = int(meta_key[-2:])
-            curr_date = time.Time(self.get_meta_value(headarr, 'mjd'), format='mjd')
+            try:
+                curr_date = time.Time(self.get_meta_value(headarr, 'mjd'), format='mjd')
+            except:
```

Comment:
use the error thrown, not a full blown `except`

Reply:
I would usually agree, except that pypeit_setup should never crash and this line is added since it was crashing. I actually do want to test all possible exceptions here since I really don't want pypeit_setup to ever crash. IIf you want to go through and figure out all the possible ways this could crash and test on those errors please go ahead. Indeed it was a failure to consider all those failure modes that was leading to the crashes in pypeit_setup when this line was added to decide on the detector to use.

Reply:
This routine is called by more than `pypeit_setup`.
  And I am pretty certain the fix here will crash for LRIS data older than 2014 (with bad headers).

I think I see the issue and am going to submit a PR to this branch as a fix.

Reply:
Another option here would be to make the lamps (and MJD) not required.  But I don't recommend that.